src/com/amazon/ion/impl/lite/IonSymbolLite.java

```diff
@@ -291,7 +289,13 @@ final void writeBodyTo(IonWriter writer, SymbolTableProvider symbolTableProvider
         // we throw (cannot serialize) or do we pass the sid thru???
 
         // NB! This will throw if symbol is not set
-        writer.writeSymbolToken(symbolValue(symbolTableProvider));
+        String text = _stringValue(symbolTableProvider);
+        if (text != null) {
+            writer.writeSymbol(text);
+        } else {
+            writer.writeSymbolToken(_Private_Utils.newSymbolToken(getSymbolId(symbolTableProvider)));
+        }
```

Comment:
Avoiding unnecessary SymbolToken allocations throughout this change resulted in speedups of up to 38%.